Prepares Flink ML for 2.0.0 release

lindong28 opened the pull request:
## What is the purpose of the change

The purpose of this PR is to prepares Flink ML for 2.0.0 release.

## Brief change log

This PR makes the following changes:
- Adds scripts under tools/releasing
- Updates Flink ML version to 2.0-SNAPSHOT
- Includes flink-ml-uber in the pom.xml
- Updates flink-ml-uber/pom.xml to include the core module and the iteration module
- Updates the release profile in pom.xml to follow flink-statefun's practice
- Removes flink-ml-examples

## Verifying this change

This PR has been verified using the following steps:
- `mvn clean package` passes.
- Ran `jar tf ./flink-ml-uber/target/flink-ml-uber_2.11-2.0-SNAPSHOT.jar` and verified that the flink-ml-uber jar includes exactly the classes from `flink-ml-core`, `flink-ml-iteration` and `dev.ludovic.netlib`.
- Followed the commands specified [here](https://cwiki.apache.org/confluence/display/FLINK/Creating+a+Flink+Stateful+Functions+Release) and verified that the scripts added in this PR has the expected behavior.

## Does this pull request potentially affect one of the following parts:

- Dependencies (does it add or upgrade a dependency): (no)
- The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (N/A)

lindong28 commented:
@gaoyunhaii Could you help review this PR?

gaoyunhaii commented on tools/releasing/create_python_sdk_release.sh:
```diff
@@ -0,0 +1,81 @@
+RELEASE_DIR=${PROJECT_ROOT}/release/
+mkdir -p ${RELEASE_DIR}
+
+cd ${PROJECT_ROOT}/flink-ml-python-sdk/
```
This might need to be modified ?

lindong28 replied:
After going through the `create_python_sdk_release.sh` used by flink-statefun, I realize that the development of this script depends a lot on the directory structure of flink-ml-python. Since I am not the developer of flink-ml-python and I have not experience with python release, it is probably better for someone else to create this script for Flink ML.

I have removed this script from this PR.

gaoyunhaii commented on tools/releasing/create_python_sdk_release.sh:
```diff
@@ -0,0 +1,81 @@
+cd ${PROJECT_ROOT}/flink-ml-python-sdk/
+./build-distribution.sh
+
+mv dist/* ${RELEASE_DIR}
```
We might also not have dist yet.

gaoyunhaii commented on tools/releasing/update_branch_version.sh:
```diff
@@ -0,0 +1,73 @@
+mvn versions:set -DgenerateBackupPoms=false -DnewVersion=${NEW_VERSION}
+
+# change version in Python SDK's setup.py file
+perl -pi -e "s#version=\'$OLD_VERSION\'#version=\'$NEW_VERSION\'#" flink-ml-python/setup.py
```
We might need also modify this line to 
```
SHORT_OLD_VERSION=${OLD_VERSION%%"-SNAPSHOT"}
SHORT_NEW_VERSION=${NEW_VERSION%%"-SNAPSHOT"}
perl -pi -e "s#version=\'$SHORT_OLD_VERSION\'#version=\'$SHORT_NEW_VERSION\'#" flink-ml-python/pyflink/ml/version.py
```

lindong28 replied:
After going through the create_python_sdk_release.sh used by flink-statefun, I realize that the development of this script depends a lot on the directory structure of flink-ml-python. Since I am not the developer of flink-ml-python and I have not experience with python release, it is probably better for someone else to create this script for Flink ML.

I have removed this script from this PR.

lindong28 replied:
I have updated this line to the following lines, similar to the approach used by `flink/tools/releasing/update_branch_version.sh`.

```
perl -pi -e "s#^__version__ = \".*\"#__version__ = \"${NEW_VERSION}\"#" flink-ml-python/pyflink/ml/version.py
perl -pi -e "s#-SNAPSHOT#\\.dev0#" flink-ml-python/pyflink/ml/version.py
```

lindong28 commented:
@gaoyunhaii Thanks for the review! I have updated PR and addressed all comments.

I have tested the scripts added in this PR by executing the commands specified in https://cwiki.apache.org/confluence/display/FLINK/Creating+a+Flink+Stateful+Functions+Release. Could you take another look at the PR?

gaoyunhaii commented:
Very thanks @lindong28 for the review! LGTM and would merge~
